File: kirun-java/src/test/java/com/fincity/nocode/kirun/engine/runtime/util/string/StringFormatterTest.java
package com.fincity.nocode.kirun.engine.runtime.util.string;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

import com.fincity.nocode.kirun.engine.util.string.StringFormatter;

class StringFormatterTest {

	@Test
	void test() {
		
		assertEquals("Hello Kiran", StringFormatter.format("Hello $", "Kiran"));
		assertEquals("$Hello Kiran", StringFormatter.format("\\$Hello $", "Kiran"));
		assertEquals("Hi Hello How are you $?", StringFormatter.format("Hi Hello How are you $?"));
		assertEquals("Hi Hello How are you 123$", StringFormatter.format("Hi Hello How are you $$$$", "1", "2", "3"));
		assertEquals("Hi Hello How are you $123", StringFormatter.format("Hi Hello How are you \\$$$$", "1", "2", "3"));
		assertEquals("Hi Hello How are you 12$$", StringFormatter.format("Hi Hello How are you $$$\\$", "1", "2"));
	}

}

File: kirun-js/__tests__/engine/util/string/StringFormatterTest.ts
import { StringFormatter } from '../../../../src/engine/util/string/StringFormatter';

test('StringFormatter Test', () => {
    expect(StringFormatter.format('Hello $', 'Kiran')).toBe('Hello Kiran');
    expect(StringFormatter.format('\\$Hello $', 'Kiran')).toBe('$Hello Kiran');
    expect(StringFormatter.format('Hi Hello How are you $?')).toBe('Hi Hello How are you $?');
    expect(StringFormatter.format('Hi Hello How are you $$$$', '1', '2', '3')).toBe(
        'Hi Hello How are you 123$',
    );
    expect(StringFormatter.format('Hi Hello How are you \\$$$$', '1', '2', '3')).toBe(
        'Hi Hello How are you $123',
    );
    expect(StringFormatter.format('Hi Hello How are you $$$\\$', '1', '2')).toBe(
        'Hi Hello How are you 12$$',
    );
    expect(StringFormatter.format('Extra closing $ found', '}')).toBe('Extra closing } found');
});
